textalyze.js

```diff
@@ -11,24 +11,14 @@
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
-function itemCounts(array) {
-  let counts = new Map();
-
-  array.forEach(item => {
-
-    if (!counts.has(item)) {
-      counts.set(item, 0);
-    }
-
-    counts.set(item, counts.get(item) + 1);
-
-  });
-
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
-
-  return counts;
-}
+const itemCounts = array => array
+  .reduce((counts, item) => {
+    !counts.has(item)
```

Comment:
Nice!

Check out what [Map.prototype.set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set) returns, though.  It returns the updated `Map` object directly.

So:

```javascript
!counts.has(item) ? counts.set(item, 1) : counts.set(item, counts.get(item) + 1);
return counts
```

Can become

```javascript
return !counts.has(item) ? counts.set(item, 1) : counts.set(item, counts.get(item) + 1);
```

Reply:
BTW, unless I say something needs to be fixed, feel free to merge the PR into your code.  I will always find a way to give feedback.  You can submit a new PR for the refactoring and request feedback on that, too.